docs/hooks-script.md

```diff
@@ -0,0 +1,231 @@
+# The ScriptAI system
+
+The ScriptAI system implemented by AC uses a special [Observer pattern](https://en.wikipedia.org/wiki/Observer_pattern) strategy to implement an event-driven programming which is also the **CORE** of our modular system.
+
+This guide, together with the our [module system](Create-a-Module.md) lets you to extend the AzerothCore without patching the core directly. This allows you to update your core by keeping your additions and customizations conflict-free!
```

Comment:
I feel like the word `core` is used twice in a row but it refers to a slightly different context, not sure how to address this though